bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/update/UpdateChannelInstructionImpl.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.thing.internal.update;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+
+/**
+ * The {@link UpdateChannelInstructionImpl} implements a {@link ThingUpdateInstruction} that updates a channel of a
+ * thing.
+ * <p />
+ * Parameters are:
+ * <ul>
+ * <li>channelId - the id of the channel</li>
+ * <li>channelTypeUID - the {@link ChannelTypeUID} of the channel</li>
+ * <li>label - the (optional) label for the channel</li>
+ * <li>description - the (optional) description for the channel</li>
+ * </ul>
+ * <p />
+ * If optional parameters are not given, they are inherited from the channel-type
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class UpdateChannelInstructionImpl implements ThingUpdateInstruction {
+    private final int thingTypeVersion;
+    private final List<String> parameters;
+    private final boolean addOnly;
+
+    UpdateChannelInstructionImpl(int thingTypeVersion, List<String> parameters, boolean addOnly) {
+        this.thingTypeVersion = thingTypeVersion;
+        this.parameters = parameters;
+        this.addOnly = addOnly;
+    }
+
+    @Override
+    public int getThingTypeVersion() {
+        return thingTypeVersion;
+    }
+
+    @Override
+    public void perform(Thing thing, ThingBuilder thingBuilder) {
+        ChannelUID affectedChannelUid = new ChannelUID(thing.getUID(), parameters.get(0));
+        if (!addOnly) {
+            thingBuilder.withoutChannel(affectedChannelUid);
```

Comment:
That's a good question. It might be very difficult to get it right here. If the `ChannelType` changes, there is no guarantee that the "new" `ChannelType` requires the same configuration as the old one. Maybe we could apply a policy here: If the configuration shall be copied, use `UPDATE_CHANNEL`, if it shall be discarded, use `REMOVE_CHANNEL` and `ADD_CHANNEL`. WDYT?

Reply:
Yes, that would make sense.